pkg/kapp/diff/stripnamehashsuffix.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"strings"
+
+	ctlconf "github.com/vmware-tanzu/carvel-kapp/pkg/kapp/config"
+	ctlres "github.com/vmware-tanzu/carvel-kapp/pkg/kapp/resources"
+)
+
+type stripNameHashSuffixConfig struct {
+	ResourceMatcher ctlres.ResourceMatcher
+}
+
+func (d stripNameHashSuffixConfig) EnabledFor(res ctlres.Resource) (stripEnabled bool) {
+	return d.ResourceMatcher.Matches(res)
+}
+
+func newStripNameHashSuffixConfig(resourceMatchers [][]ctlres.ResourceMatcher) (result stripNameHashSuffixConfig) {
+	result = stripNameHashSuffixConfig{}
+
+	var allMatchers []ctlres.ResourceMatcher
+	for _, matchers := range resourceMatchers {
+		// configurations that do not specify any matchers (like default
+		// config) have nil, which would lead AndMatcher to never match.
+		// as all the configs AndMatchers are ANDed, too, this would prevent
+		// matchers from any configuration to match, which would strip the
+		// whole suffix-strip configuration of any usefulness; as such we
+		// exclude nil matchers. :)
+		if matchers != nil {
```

Comment:
I think we can skip this. I do not think that anyone would want to add an empty config to the kapp config